R/readSource.R

```diff
@@ -4,41 +4,41 @@
 #' wrapper for specific functions designed for the different possible source
 #' types.
 #'
-#' @param type source type, e.g. "IEA". A list of all available source types
-#' can be retrieved with function \code{\link{getSources}}.
-#' @param subtype For some sources there are subtypes of the source, for these
-#' source the subtype can be specified with this argument. If a source does not
+#' @param type A character string referring to the source type, e.g. "IEA" which would
+#' internally call a function called `readIEA` (the "wrapped function"). A list of
+#' available source types can be retrieved with function \code{\link{getSources}}.
+#' @param subtype A character string. For some sources there are subtypes of the source, for these
+#' sources the subtype can be specified with this argument. If a source does not
 #' have subtypes, subtypes should not be set.
-#' @param subset Similar to \code{subtype} a source can also have \code{subsets}. A \code{subsets}
-#' can be used to only read part of the data. This can in particular make sense for huge
-#' data sets where reading in the whole data set might be impractical or even
-#' infeasible.
+#' @param subset A character string. Similar to \code{subtype} a source can also have \code{subsets}.
+#' A \code{subsets} can be used to only read part of the data. This can in particular make sense
+#' for huge data sets where reading in the whole data set might be impractical or even infeasible.
 #' @param convert Boolean indicating whether input data conversion to
 #' ISO countries should be done or not. In addition it can be set to "onlycorrect"
 #' for sources with a separate correctXXX-function.
-#' @return magpie object with the temporal and data dimensionality of the
-#' source data. Spatial will either agree with the source data or will be on
-#' ISO code country level depending on your choice for the argument "convert".
-#' @author Jan Philipp Dietrich, Anastasis Giannousakis, Lavinia Baumstark
+#' @return The read-in data, usually a magpie object. Technically speaking
+#' `if (is.list(result)) result$x else result` will be returned, where `result` is the
```

Comment:
not sure if that part is easy to understand for everyone

Reply:
Yeah I know, I was really struggling to find a good explanation. Do you have a suggestion?

Reply:
I would just skip the sentence starting with `technically speaking`

Reply:
I think it's important to tell the user that if they return a list from their read function they will not get that list when calling `readSource`, unless they pass `supplementary = TRUE`

Reply:
I think now that `supplementary` is available and documented it's clear without the sentence, so I removed it as you suggested.